tensorrt_llm/_torch/speculative/utils.py

```diff
@@ -111,3 +113,26 @@ def get_spec_worker(spec_config, mapping):
         return Eagle3OneModelWorker(spec_config, mapping)
     else:
         return None
+
+
+def get_draft_model_prompt(spec_dec_mode: SpeculativeDecodingMode,
+                           input_tokens: torch.Tensor) -> torch.Tensor:
+    """
+    Can be used to modify prompts for speculative algorithms that need to update tokens
+    before drafting.
+    """
+    if spec_dec_mode.is_eagle3():
+        # EAGLE3 always throws away the first token when processing draft inputs
+        return input_tokens[1:]
+    return input_tokens
+
+
+def get_num_extra_kv_tokens(spec_config):
+    """
+    Implementation detail for one model implementations of speculative decoding. Extra
+    KV cache tokens are required.
+    """
+    if spec_config.spec_dec_mode.is_eagle3_one_model(
+    ) or spec_config.spec_dec_mode.is_mtp():
```

Comment:
For MPT, only `MTP_Eagle` has `num_extra_kv_tokens > 0`.
```suggestion
    ) or spec_config.spec_dec_mode.is_mtp_eagle():
```